knox/views.py

```diff
@@ -25,15 +31,16 @@ def post(self, request, format=None):
                     {"error": "Maximum amount of tokens allowed per user exceeded."},
                     status=status.HTTP_403_FORBIDDEN
                 )
-        token = AuthToken.objects.create(request.user)
+        token_ttl = self.get_token_ttl()
+        token = AuthToken.objects.create(request.user, token_ttl)
```

Comment:
it's already doing that by default:
https://github.com/James1345/django-rest-knox/blob/master/knox/models.py#L12